### DISCUSSION TITLE
Ubuntu 20.04 Focal does not install automatically

### DISCUSSION THREAD

**Discussion by pbvafer** [cobbler/cobbler#3143] [Version specific questions]
Version: Cobbler 3.3.2

@SchoolGuy Hi, I have a new problem, this time RHEL8 goes very well, now I'm testing with Ubuntu 20.04.4 focal, and when I generate the profile and then the Cobbler system generates a url (http://192.168.60.10/cblr/svc/op/autoinstall/system/nodo1) where the ks file is supposed to be, at least in RHEL8 it works and I see my ks, in ubuntu I get this:

```
# this file intentionally left blank
# admins:  edit it as you like, or leave it blank for non-interactive install
```

**Comment by pbvafer** [comment]
Mi Ubuntu ks file:

```
lang en_US.UTF-8
keyboard us
text
skipx
user --name=name --password=psswd
reboot
url --url=http://192.168.60.10/cblr/links/Ubuntu-20_04-casper-x86_64/
network --bootproto=static --ip=192.168.60.21 --netmask=255.255.255.0 --gateway=192.168.60.1 --nameserver=192.168.60.1,8.8.8.8 --device=enp0s25 --hostname=nodo1-ngl --onboot=on --noipv6 --activate
rootpw --plaintext passwd
firewall --disabled
authselect --enableshadow --passalgo=sha512
selinux --permissive
timezone --utc Europe/Madrid
zerombr
clearpart --all --drives=sdb --initlabel
bootloader --location=mbr --driveorder=sdb
autopart --type=lvm
firstboot --enable

%packages
ubuntu-minimal
openssh-server
screen
curl
wget
acpid
unattended-upgrades
linux-image-generic
python-apt
lshw
lldpd
dmidecode
%end
```

**Reply by pbvafer** [reply]
I haven't done anything different with the Ubuntu installation than with RHEL8, I understand it should work in a similar way. This is the file that Cobbler generates automatically to launch PXE:

```
timeout 1
prompt 0
default nodo1
ontimeout nodo1
LABEL nodo1
        MENU LABEL nodo1
        kernel /images/Ubuntu-20_04-casper-x86_64/vmlinuz
        append initrd=/images/Ubuntu-20_04-casper-x86_64/initrd  auto-install/enable=true priority=critical netcfg/choose_interface=auto url=http://192.168.60.10/cblr/svc/op/autoinstall/system/nodo1 hostname=nodo1 domain=local.lan suite=focal
        ipappend 2
```

**Reply by pbvafer** [reply]
And the Ubuntu installation stops here:
![IMG_20220606_121202](https://user-images.githubusercontent.com/79220152/172141621-fd683965-8aa3-4884-9a65-dfe6358c1590.jpg)

**Reply by SchoolGuy** [reply]
@pbvafer What is your value for `autoinstall` of the system? Because the default is `default.ks` and that has exactly that content. If you would set this for example to `sample.seed`, then you should already get further down the road.

**Reply by pbvafer** [reply]
I haven't touched the `autoinstall` option, I understand that for ubuntu you use the `sample.seed` and not the `default.ks`, is that right?

**Reply by SchoolGuy** [reply]
Yes for example. You could also put your preseed file in the autoinstall template directory and then set the filename of your preseed file.

**Reply by pbvafer** [reply]
This is the command I used to create the cobbler profile, using the ks file above.
`cobbler profile add --name=Ubuntu-20_04-Nodo1 --distro=Ubuntu-20_04-casper-x86_64 [--autoinstall=ubuntu-20_04.ks]`

**Reply by SchoolGuy** [reply]
Are the square brackets intentionally or are they not part of the command?

**Reply by pbvafer** [reply]
Mmm now that you mention it I put it as is, good point.

**Reply by pbvafer** [reply]
With the square brackets the flag is skipped and it's as if I didn't put anything, right?

**Reply by SchoolGuy** [reply]
Depends on the shell but most likely yes.

**Reply by pbvafer** [reply]
I'm not in the office right now, tomorrow morning I'll try it and confirm, thank you very much for your attention!

**Reply by SchoolGuy** [reply]
I would recommend doing this: `cobbler profile edit --name="Ubuntu-20_04-Nodo1" --autoinstall="ubuntu-20_04.ks"`

I would always put double colons around the argument to prevent globbing of the arguments.

**Reply by pbvafer** [reply]
Good advice, thank you

**Reply by pbvafer** [reply]
After modifying the bracketed error in `autoinstall`, it is still stuck in the same place as in the screenshot above. You mentioned yesterday the `sample.seed` file, should I take it as a reference and replace my `ubuntu.ks`?

**Reply by SchoolGuy** [reply]
No if your `ubuntu.ks` file is valid then it is better to use this one. What does the `curl http://192.168.60.10/cblr/svc/op/autoinstall/system/nodo1` yield? Is the result now different?

**Reply by pbvafer** [reply]
Yes, now I see my kickstart file

**Reply by pbvafer** [reply]
But it stops at the same place, cannot find the URL.

**Reply by SchoolGuy** [reply]
Do you see what the system tries to fetch. In the picture you posted I was not able to identify the URL.

**Reply by SchoolGuy** [reply]
I see that you have the url parameter set in the append line for the kernel though.

**Reply by pbvafer** [reply]
It does not give more data, I am like in the image.

**Reply by SchoolGuy** [reply]
Does your server by any chance have multiple network interfaces? Because it could be that it tries to fetch the file via the wrong interface?

**Reply by pbvafer** [reply]
Entering the network parameters by hand:
![IMG_20220607_100454](https://user-images.githubusercontent.com/79220152/172329320-76508757-b4ca-4f6e-a480-c8ca69502a63.jpg)

**Reply by pbvafer** [reply]
I have only one interface, it should not fail as I do the same as with RHEL8.

**Reply by SchoolGuy** [reply]
So the problem is that your system is unable to find the medium with the live file system. Thus the error and the fallback to the initramfs. Sadly I never personally have installed an Ubuntu System via Preseed so I am out of ideas. I will try to make time but it may take a while until I can replicate your setup. I lack the hardware to test this atm sadly.

**Reply by pbvafer** [reply]
I will try modifying the `sample.seed`, thanks.

**Reply by pbvafer** [reply]
@SchoolGuy I have one more question, I have changed the Ubuntu ISO file to a server without GUI, and I have regenerated everything related to Cobbler, to see if the bug was in the configuration. Right now I'm DHCP booting the computer, it loads the `vmlinuz` and the `initrd`, but it doesn't advance in the installation, it loads the two files (... ok) and restarts constantly, do you know what could be happening?

**Reply by SchoolGuy** [reply]
No. Normally most OSes I know default then to a manual installation. Normally once GRUB/PXElinux has booted the kernel and intrd you should get more descriptive error messages.

**Reply by pbvafer** [reply]
@SchoolGuy Hi! Sorry to ask, do you know of any article or link to install Ubuntu 18.04, 20.04 and it would be wonderful if there were on Windows 10 on Cobbler?

**Reply by pbvafer** [reply]
It seems that to install Ubuntu you have to choose a legacy ISO and not live, is it possible? And the little I've seen about Windows is that you can't boot via PXE.

**Reply by SchoolGuy** [reply]
@pbvafer So Windows Support is present in Cobbler. I don't have an up-to-date article available but an outdated one can be found [here](https://cobbler.github.io/blog/2020/12/04/wingen.html). So PXE booting windows is not a problem at all. And yes you need to choose legacy Ubuntu for now. I have cloud-init on my list but I was not able to implement it yet.
